Dexbridge and pebble suport

StephenBlackWasAlreadyTaken opened the pull request.

jstevensog committed:
Full DexBridge Support added.

jstevensog committed:
removed com.androud.support-v4 dependency in build.gradle.

jstevensog committed:
Added settings values for Dexbridge.  This was missed in the initial merge and subsequent code corrections.

jstevensog committed:
More changes to get Dexbridge support properly into the code.

jstevensog committed:
Working changes to get Dexbridge support in.

jstevensog committed:
Fully working changes for Dexbridge support and cgm-pebble-offline.

jstevensog committed:
Fixed bridge battery updating in Home.

jstevensog committed:
Fixed PebbleSync bgDelta calculation (same now as older Dexdrip code).
Implemented deeper process checks in CollectionServiceStarter and DexCollectionService.
Lot's of other small mods that I cannot remember why I did them.
Tested and working fine except for late packets causing double dots.  Looking into solutions.

jstevensog committed:
Increased setRetryTimer time in DexCollectionService to remove "stacked" dots.
Made PebbleSync use phone battery to display if dexbridge is not being used.

jstevensog committed:
Ensured timestamps propagated through from DexCollectionService to TransmitterData and BgReading classes.  May fix stacked dots issue.

jstevensog committed:
Minor changes to get rid of warnings, and make more code efficiencies

jstevensog committed:
Added missing file 27.sql.

StephenBlackWasAlreadyTaken commented:
oh goodness, this is off tzachis branch rather than master

jstevensog commented:
Sorry.  I thought it was off master.  I did merge in the former
DexbridgeSupport, so perhaps it came through with that,
Cheers

On Thu, Apr 23, 2015 at 10:26 PM, Stephen Black notifications@github.com
wrote:

> oh goodness, this is off tzachis branch rather than master
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-95568380
> .

## 

John Stevens
"You are how you live, not what you have."

StephenBlackWasAlreadyTaken commented:
No worries! Just was surprised, it's actually in pretty good shape and will
probably be merged in right after alerts anyway! It just makes it harder to
see your changes vs the alert code changes but I actually cherry picked
them our to view locally
On Apr 23, 2015 7:41 PM, "John" notifications@github.com wrote:

> Sorry. I thought it was off master. I did merge in the former
> DexbridgeSupport, so perhaps it came through with that,
> Cheers
> 
> On Thu, Apr 23, 2015 at 10:26 PM, Stephen Black notifications@github.com
> wrote:
> 
> > oh goodness, this is off tzachis branch rather than master
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-95568380
> > 
> > .
> 
> ## 
> 
> John Stevens
> "You are how you live, not what you have."
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-95749543
> .

jstevensog commented:
The main changes are in TransmitterData, DexCollectionService,
CollectionServiceStarter (minor), BgReading, BgSendQueue, Home, and of
course PebbleSync.  I also had to add some settings menu items (obviously
for TXID, which I am not sure I made conditional on selection of Dexbridge,
so maybe add that, or I will push another version soon).
You can probably simply copy PebbleSync as is, as the original is broken
anyway.
The others, well I will leave that to your judgement.
I also took the ACRA debug uploading stuff out of xDrip.   That really
should be a settings option, to opt in, not a general "feature".  I took it
out cause you would have been flooded by my app crashes when I first got
into it.
Don't think there are any other places that need much work.
I have left all the notifications stuff alone.

BTW, I reduced the fuzzer value in BgGraphBuilder to 1 minute in ms, rather
than 5 minute in ms, and the stacked dots have disappeared.  Looks a lot
better.  Could probably remove it altogether, or make it 10 seconds in ms,
but I have left it at this for now.
Cheers

On Fri, Apr 24, 2015 at 1:48 PM, Stephen Black notifications@github.com
wrote:

> No worries! Just was surprised, it's actually in pretty good shape and will
> probably be merged in right after alerts anyway! It just makes it harder to
> see your changes vs the alert code changes but I actually cherry picked
> them our to view locally
> On Apr 23, 2015 7:41 PM, "John" notifications@github.com wrote:
> 
> > Sorry. I thought it was off master. I did merge in the former
> > DexbridgeSupport, so perhaps it came through with that,
> > Cheers
> > 
> > On Thu, Apr 23, 2015 at 10:26 PM, Stephen Black <
> > notifications@github.com>
> > wrote:
> > 
> > > oh goodness, this is off tzachis branch rather than master
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > 
> > https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-95568380
> > 
> > > .
> > 
> > ## 
> > 
> > John Stevens
> > "You are how you live, not what you have."
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-95749543
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-95791505
> .

## 

John Stevens
"You are how you live, not what you have."

StephenBlackWasAlreadyTaken commented:
werent you saying you had stacked dots a while ago? I only added the fuzzer here fairly recently (and I keep it at 2.5 minutes nowadays). it keeps the data points spaced out much nicer on the graph, I havnt played with 1 minute yet

StephenBlackWasAlreadyTaken commented:
I wonder if its condensing it too much for your screen, what phone do you use?

jstevensog commented:
The dots stacking a while ago was because of duplicate data points being received. Fixed that in the wixel code and the DexCollectionService, with some duplicate packet detection. But think I removed that/didn't add it to xDrip as the wixel fixes were enough. The phone is a Z. Very large (not phablet) high res screen. 
Cheers

John Stevens
M - +61 418526349

> On 25 Apr 2015, at 02:00, Stephen Black notifications@github.com wrote:
> 
> I wonder if its condensing it too much for your screen, what phone do you use?
> 
> —
> Reply to this email directly or view it on GitHub.

StephenBlackWasAlreadyTaken committed:
Merge branch 'Alerts-and-Notifications' into DexbridgeAndPebbleSuport

Conflicts:
	app/app.iml
	app/build.gradle
	app/src/main/java/com/eveningoutpost/dexdrip/Home.java
	app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
	app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
	app/src/main/java/com/eveningoutpost/dexdrip/xdrip.java
	app/src/main/res/xml/pref_data_source.xml

StephenBlackWasAlreadyTaken committed:
hide txID pref when not usable

StephenBlackWasAlreadyTaken committed:
assign filtered data in normal xdrip collection

StephenBlackWasAlreadyTaken committed:
break up creates and add migration for those of us without transmitter filtered data

StephenBlackWasAlreadyTaken committed:
add migration for unclean bg

StephenBlackWasAlreadyTaken committed:
Merge branch 'Alerts-and-Notifications' into DexbridgeAndPebbleSuport

mgranberry committed:
recycle notification bitmaps and use a smaller bitmap for notification tray in order to reduce memory footprint

mgranberry commented on app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java:
```diff
@@ -39,7 +41,11 @@
     public String operation_type;
 
-    public static BgSendQueue nextBgJob() {
+    private static Context mContext = Home.getContext();
```
This context was null once for me, leading to a crash.  I suspect Home had been reaped for memory and a service kicked off BgSendQueue.  Keeping static contexts around is generally considered bad form because it leads to leaking contexts.  It looks like it is probably better to use the context already being passed into the method using pebbleSync.

mgranberry commented on app/src/main/java/com/eveningoutpost/dexdrip/Home.java:
```diff
@@ -64,16 +67,20 @@
     BroadcastReceiver _broadcastReceiver;
     BroadcastReceiver newDataReceiver;
+    BroadcastReceiver newSavedBgReceiver;
+    private static Context mContext;
```
mContext is not defined prior to the Home activity being shown, but gets referenced in getContext() and gets used in services.  If the Activity is reaped for memory pressure, the services may be restarted and the context will be invalid.  Services are their own context.  Maybe it is more appropriate to initialize BgSendQueue from the services using them or pass in a PebbleSync that was initialized with the service as a context.  Holding a reference to an application Context isn't likely to work in all circumstances.

In most cases these static-rich classes look like good candidates for a bound service and an AIDL interface.  It gets a little hairy writing the first one, but it's pretty simple and provides an API across process boundaries with well-defined contexts, etc for all of them.

jstevensog committed:
Minor changes to log messages in BgReading and MongoWrapper.
Altered pgGrapshBuilder to use a finer time granularity.

jstevensog committed:
Merge branch 'DexbridgeAndPebbleSuport' of https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental into DexbridgeAndPebbleSuport

Conflicts:
	app/app.iml
	app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java

StephenBlackWasAlreadyTaken commented:
@jstevensog it looks like you merged in master from the normal repo (and basically restarted the merge from the beginning, things were already merged to be on level with experimental master).
Let me know if you need that rebased out... I dont know if you will be able to make it work otherwise

jstevensog commented:
Yep. Rebase it out. I will archive my local repo, get the latest, and start again. 
Got it built but still waiting on a transmitter. 
Cheers

John Stevens
M - +61 418526349

> On 11 May 2015, at 22:02, Stephen Black notifications@github.com wrote:
> 
> @jstevensog it looks like you merged in master from the normal repo (and basically restarted the merge from the beginning, things were already merged to be on level with experimental master).
> Let me know if you need that rebased out... I dont know if you will be able to make it work otherwise
> 
> —
> Reply to this email directly or view it on GitHub.

jstevensog commented:
How'd you go with rebasing?  Should I take a clean copy of the branch now?
And should I just get Master, or get the DexbridgeAndPebbleSuport branch?
Thanks & Cheers

On Tue, May 12, 2015 at 7:03 AM, John Stevens jstevensog@gmail.com wrote:

> Yep. Rebase it out. I will archive my local repo, get the latest, and
> start again.
> Got it built but still waiting on a transmitter.
> Cheers
> 
> John Stevens
> M - +61 418526349
> 
> On 11 May 2015, at 22:02, Stephen Black notifications@github.com wrote:
> 
> @jstevensog https://github.com/jstevensog it looks like you merged in
> master from the normal repo (and basically restarted the merge from the
> beginning, things were already merged to be on level with experimental
> master).
> Let me know if you need that rebased out... I dont know if you will be
> able to make it work otherwise
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/11#issuecomment-100886683
> .

## 

John Stevens
"You are how you live, not what you have."

jstevensog committed:
Dexbridge Support and Pebble Support.
Note, Pebble Support is untested, in this commit, but there is unlikely to
be an issue.
Added 60 second age to lastPacketData in broadcastUpdate in
DexCollectionService.
Re-instated Pebble options in settings.
Put all Dexbridge options in Preferences.java and associated xml.

StephenBlackWasAlreadyTaken commented:
@jstevensog you need to stop making changes that are not directly related to dexbridge, this is just getting harder and harder to maintian, also pulling in individual commits is going to make this impossible to keep up, what still needs to be done directly related to dexbridge to get this ready to roll?

StephenBlackWasAlreadyTaken committed:
Merge remote-tracking branch 'origin/master' into DexbridgeAndPebbleSuport

Conflicts:
	app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
	app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java

jstevensog commented:
Hi Stephen,
That is exactly what I am trying to do. My bad for not understanding git, as I am not trying to make changes that are unrelated to dexbridge. I have a clean copy of xDrip-E as of yesterday, and have created a new branch to just do the dexbridge stuff. 
I am also finding that the ongoing rate of change is making it difficult to make the dexbridge code fit into the head.  
In order to get it in, I must make changes to Home, PebbleSync, DexCollectionService, TransmitterRawData, as a minimum. I may or may not have to alter BgReading, but what is thee currently is definitely not working. 
Happy to do what needs to be done, just need to know exactly what that is. 
I added filtered_data and a flag showing if filtered_data or raw_data were selected in calculations in BgReading a long time ago, and was not aware that these had been brought in by others. 
The problem is that we have no communication of who is working in what, what parts are sacrosanct, etc. So, in my case, I am doing what I can to get the support in. 
I need support for Dexbridge either in DexCollectionService or its own service, in TransmitterData as a minimum. I also need the Home modified, settings screens modified, and PebbleSync modified. Most of this is now done. 
But until I have a kitkat or lollipop dev phone, I cannot test. I can do without my shoddy filtering I'm BgReading IF the standard class works, which right now it won't on my Jellybean phone. 
There is not much more to to change. 
Cheers

John Stevens
M - +61 418526349

> On 14 May 2015, at 23:41, Stephen Black notifications@github.com wrote:
> 
> @jstevensog you need to stop making changes that are not directly related to dexbridge, this is just getting harder and harder to maintian, also pulling in individual commits is going to make this impossible to keep up, what still needs to be done directly related to dexbridge to get this ready to roll?
> 
> —
> Reply to this email directly or view it on GitHub.

jstevensog committed:
Save of current state.  Not ready for release.

jstevensog committed:
Merge branch 'master' of https://github.com/jstevensog/xDrip-Experimental into DexbridgeAndPebbleSupportV2

Conflicts:
	app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

jstevensog committed:
Removed any spurious code not directly related to Dexbridge or Pebble.

StephenBlackWasAlreadyTaken committed:
some minor cleanup

jstevensog committed:
Fixed but in PebbleSync.

jstevensog committed:
Corrected bgDelta in PebbleSync.

jstevensog committed:
Modified BgGraphBuilder unitized_string function to always return mmol
with 1 decimal place.  Required by Pebble Watch Face, and probably a good
idea.

jstevensog committed:
Added code in receivedDataHandler in PebbleSync to prevent duplicate data
requests from causing the watch face to buzz and light up.

jstevensog committed:
Fixed bug in BgSendQueue that was creating lots of PebbleSync.

jstevensog committed:
Altered PebbleSync and BgSendQueue to try to fix the multiple initiations
of PebbleSync.

jstevensog committed:
Final fix to BgSendQueue and PebbleSync for multiple PebbleSync issue.

StephenBlackWasAlreadyTaken committed:
reduce memory leakage

StephenBlackWasAlreadyTaken committed:
turn pebblesync into a service

StephenBlackWasAlreadyTaken committed:
start and stop pebble sync when toggled

jstevensog committed:
Added BG_SAVED listener register and unregister in onCreate and onDestroy.

jstevensog committed:
Final PebbleSync changes to run as a service.
Now starts with xDrip if broadcast_to_pebble preference is true.
Stops if broadcast_to_pebble changes to false.
Starts if broadcast_to_pebble changes to true.
Home.java sets bridge_battery prefernce to 0 in onCreate, to cause
red Waiting for packet message to appear on xDrip start.

jstevensog committed:
Bridge Battery displays strike-through when packet missed.

jstevensog committed:
Removed bridge_battery reset.  It was resetting when screen changed
orientation.  Not good.  Will work on a solution at another time.
Strikethrough will display if xDrip has been down for longer than the
"missed signal" time.  That should do for now.

jstevensog committed:
Modified to resolve merge conflict with master.

StephenBlackWasAlreadyTaken committed:
Merge remote-tracking branch 'origin/DexbridgeAndPebbleSupportV2' into DexbridgeAndPebbleSuport

Conflicts:
	app/app.iml
	app/src/main/java/com/eveningoutpost/dexdrip/Home.java
	app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
	app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java
	app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java
	app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java
	app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgSendQueue.java
	app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/CollectionServiceStarter.java
	app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/PebbleSync.java
	app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java
	app/src/main/res/layout/activity_home.xml
	app/src/main/res/xml/pref_advanced_settings.xml

StephenBlackWasAlreadyTaken committed:
cleanup after merge
